### PR TITLE
First pass at adding Honeycomb beeline

### PR DESCRIPTION
**Pull request by tfhartmann**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -19,6 +19,9 @@ ENV CONSUL_PREFIX=healthcheck
 ENV CONSUL_HTTP_ADDR=""
 ENV CONSUL_HTTP_AUTH=""
 ENV EXTRA_ARGS=""
+#ENV HONEYCOMB_API_KEY="ce8f58cf494fa93b7376d053c577a894"
```
Comment (hakamadare) [inline Dockerfile]: is this an actual API key?  perhaps don’t check into GH 😄

File: main.go [hunk 2]
```diff
@@ -7,18 +7,24 @@ import (
 
 	"github.com/caarlos0/env"
 	"github.com/urfave/negroni"
+
+  beeline "github.com/honeycombio/beeline-go"
```
Comment (hakamadare) [inline main.go:11]: i think there’s some weird tab/whitespace handling going on here. consider using [go fmt](https://golang.org/cmd/go/#hdr-Gofmt__reformat__package_sources)?

File: main.go [hunk 3]
```diff
@@ -7,18 +7,24 @@ import (
...
 type config struct {
-	Port        int    `env:"PORT"`
-	Application string `env:"HEALTHCHECK_APP" envDefault:"healthcheck"`
-	Status      int    `env:"HEALTHCHECK_STATUS" envDefault:"200"`
-	Metrics     string `env:"HEALTHCHECK_METRICS" envDefault:"{}"`
+	Port         int    `env:"PORT"`
+	Application  string `env:"HEALTHCHECK_APP" envDefault:"healthcheck"`
+	Status       int    `env:"HEALTHCHECK_STATUS" envDefault:"200"`
+	Metrics      string `env:"HEALTHCHECK_METRICS" envDefault:"{}"`
+	HoneyKey     string `env:"HONEYCOMB_API_KEY" envDefault:"ce8f58cf494fa93b7376d053c577a894"`
```
Comment (hakamadare) [inline main.go]: see above re: checking in API key

File: main.go [hunk 4]
```diff
@@ -27,10 +33,19 @@ type healthcheck struct {
...
 	cfg := config{}
+  env.Parse(&cfg)
+
 	hc := healthcheck{}
 
 	// default config for our own healthcheck
-	defcfg := config{Application: "healthcheck", Status: 200, Metrics: "{}"}
+	defcfg := config{Application: "healthcheck", Status: 200, Metrics: "{}" }
+
+	beeline.Init(beeline.Config{
```
Comment (hakamadare) [inline main.go:43]: maybe consider breaking the config out into its own variable, to make it easier to parse?

```go
beeCfg := beeline.Config{
    WriteKey: cfg .HoneyKey,
    ...
}

beeline.Init(beeCfg)
```

**Review by hakamadare — COMMENTED** [review]
LGTM!  some feedback in comments

File: Dockerfile (hunk 1 above)
Comment (tfhartmann) [reply Dockerfile]: um... no, that couldn't be valid key... nope, nothing to see here. Thanks for point that out, I revoked that key.

File: main.go (hunk 3 above)
Comment (tfhartmann) [reply main.go]: I'm an idiot

**Comment by hakamadare** [comment]
sweet!  well done. remember to squash and merge to get rid of the api key 😄

File: main.go (hunk 4 above)
Comment (hakamadare) [reply main.go:43]: oh, also, you might consider checking that `cfg.HoneyKey` looks like a valid API key, and only `beeline.Init()` if it is?  i looked at the beeline docs and it’s not clear to me how they handle invalid configs

**Review by hakamadare — APPROVED** [review]

**Pull request merged**